spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/RabbitConnectionFactoryBean.java

```diff
@@ -162,6 +162,11 @@ public class RabbitConnectionFactoryBean extends AbstractFactoryBean<ConnectionF
 	private String trustStoreAlgorithm = SUN_X509;
 
 	public RabbitConnectionFactoryBean() {
+		this(new ConnectionFactory());
+	}
+
+	public RabbitConnectionFactoryBean(ConnectionFactory connectionFactory) {
```

Comment:
This is not correct usage of the factory pattern: if you have `com.rabbitmq.client.ConnectionFactory` instance, then you don't need `RabbitConnectionFactoryBean` at all.
Also: we follow the rule do not mutate externally provided objects.

So, I cannot accept this PR altogether.

Will keep it open until your answer.

Thanks for understanding.